From 63aaa5edcfc89cc8490055ace4725b98e79cd036 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Tue, 17 Jul 2018 10:43:32 +0800
Subject: [PATCH 4499/5242] MLK-18935 usb: host: xhci-plat: add platform TPL
 support

commit  b4fe05937eba1c4491fb5974374574f8a7c677d9 from
https://source.codeaurora.org/external/imx/linux-imx.git

The TPL support is used to identify targeted devices during
EH2.0 and EH3.0 certification test.

Acked-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
(cherry picked from commit a2525e0c5f3ad5e07f91e65fef7eb02c32d69c70)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/host/xhci-plat.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 42b699b..913f9067 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -19,6 +19,7 @@
 #include <linux/slab.h>
 #include <linux/acpi.h>
 #include <linux/busfreq-imx.h>
+#include <linux/usb/of.h>
 
 #include "xhci.h"
 #include "xhci-plat.h"
@@ -309,6 +310,8 @@ static int xhci_plat_probe(struct platform_device *pdev)
 	}
 
 	request_bus_freq(BUS_FREQ_HIGH);
+	hcd->tpl_support = of_usb_host_tpl_support(sysdev->of_node);
+	xhci->shared_hcd->tpl_support = hcd->tpl_support;
 
 	ret = usb_add_hcd(hcd, irq, IRQF_SHARED);
 	if (ret)
-- 
1.7.9.5

